agent/src/agent/pipeline/config/jython_scripts/jdbc.py

```diff
@@ -30,6 +31,36 @@ def to_timestamp(date):
     return int((date - epoch).total_seconds())
 
 
+def check_missing_intervals(cur_offset, interval):
+    db_offset = get_db_offset()
+    if not db_offset:
+        return
+    while ((cur_offset - db_offset) / interval.total_seconds()) > 1:
+        sdc.log.debug(
+            'PROCESSING MISSED DATA: '
+            'from ' + str(db_offset) + ' to ' + str(db_offset + interval.total_seconds())
+        )
```

Comment:
in a situation where there is missing data for one day but data for the next day is present the pipeline will be stuck because the query interval is always 1 day. So in this case `date from` should be the db offset but `date to` should be the end of the current interval.